shared/fs/popups/row-action-popup-container.js

```diff
@@ -0,0 +1,160 @@
+          dispatch(
+            navigateAppend([
+              {
+                props: {path, isShare: true},
+                selected: 'transferPopup',
+              },
+            ])
+          )
+        },
+        shareNative: (path: Types.Path) => {
+          dispatch(FSGen.createDownload({path, intent: 'share'}))
+          dispatch(
+            navigateAppend([
+              {
+                props: {path, isShare: true},
+                selected: 'transferPopup',
+              },
+            ])
+          )
+        },
+        share: (path: Types.Path) =>
+          dispatch(
+            navigateAppend([
+              {
+                props: {path, isShare: true},
+                selected: 'rowAction',
+              },
+            ])
+          ),
+      }
+    : {
+        showInFileUI: (path: Types.Path) =>
+          dispatch(FSGen.createOpenInFileUI({path: Types.pathToString(path)})),
+      }),
+
+  ...(!isMobile || isAndroid
+    ? {
+        download: (path: Types.Path) => dispatch(FSGen.createDownload({path, intent: 'none'})),
+      }
+    : {}),
+})
+
+const getRootMenuItems = (stateProps, dispatchProps) => {
+  const {path, pathItem, fileUIEnabled} = stateProps
+  const {showInFileUI, saveImage, share, download} = dispatchProps
+  let menuItems = []
+  if (!isMobile && fileUIEnabled) {
+    menuItems.push({
+      title: 'Show in ' + fileUIName,
+      onClick: () => showInFileUI(path),
+    })
+  }
+  if (isMobile) {
+    if (Constants.isImage(pathItem.name)) {
+      menuItems.push({
+        title: 'Save',
+        onClick: () => saveImage(path),
+      })
+    }
+    menuItems.push({
+      title: 'Share...',
+      onClick: () => share(path),
+    })
+  }
+  if (!isMobile || isAndroid) {
+    // TODO make android download work
+    menuItems.push({
+      title: 'Download a copy',
+      onClick: () => download(path),
+    })
+  }
+  return menuItems
+}
+
+const getShareMenuItems = ({path}, {shareNative}) =>
+  isMobile
+    ? [
+        {
+          title: 'Send to other app',
+          onClick: () => shareNative(path),
+        },
+      ]
+    : []
+
+const mergeProps = (stateProps, dispatchProps) => {
+  const {path, pathItem, isShare, _username, childrenFolders, childrenFiles} = stateProps
+  const {onHidden} = dispatchProps
+  const itemStyles = Constants.getItemStyles(Types.getPathElements(path), pathItem.type, _username)
+  const menuItems = isShare
+    ? getShareMenuItems(stateProps, dispatchProps)
+    : getRootMenuItems(stateProps, dispatchProps)
+  return {
+    type: pathItem.type || 'unknown',
+    lastModifiedTimestamp: pathItem.lastModifiedTimestamp,
+    lastWriter: pathItem.lastWriter.username,
+    name: pathItem.name,
+    size: pathItem.size,
+    childrenFolders,
+    childrenFiles,
+    itemStyles,
+    menuItems,
+    onHidden,
+  }
+}
+
+export default compose(connect(mapStateToProps, mapDispatchToProps, mergeProps), setDisplayName('Popup'))(
+  Popup
```

Comment:
IMO it's clearer to do:
```javascript
export default compose(
  connect(mapStateToProps, mapDispatchToProps, mergeProps),
  setDisplayName('Popup')
)(Popup)
```

Reply:
Ugh it's the linter that made it one-liner.